### PR TITLE
Set additional metadata in Upload-Metadata header

### PR DESCRIPTION
**Pull request by SamsamBabadi**
According to https://tus.io/protocols/resumable-upload.html#upload-metadata the Upload-Metadata header could contain multiple items.

### CODE REVIEW

File: src/Tus/Client.php [hunk 1]
```diff
@@ -172,6 +175,79 @@ public function getChecksum() : string
...
+    }
+
+    /**
+     * Get metadata.
+     *
+     * @return array
+     */
+    public function getMetadata() : array
+    {
+        $this->addMetadata('filename', $this->fileName);
```
Comment (ankitpokhrel) [inline src/Tus/Client.php]: why do we need to add metadata inside `getMetadata` method?

Comment (SamsamBabadi) [reply src/Tus/Client.php]: For insure metadata always has `filename`, but it can check if `filename` not exist then add.

**Comment by SamsamBabadi** [comment]
Oops. my mistake, I will fix those

File: src/Tus/Client.php (hunk 1 above)
Comment (ankitpokhrel) [reply src/Tus/Client.php]: IMO It should not be `getMetadata` method that would add a `filename` meta. It should simply return metadata. I think it is more meaningful to add that filename meta as soon as we get the filename. Somewhere around [here](https://github.com/ankitpokhrel/tus-php/blob/master/src/Tus/Client.php#L87-L89).

Comment (SamsamBabadi) [reply src/Tus/Client.php]: It sounds better, however it just set in [here](https://github.com/ankitpokhrel/tus-php/blob/master/src/Tus/Client.php#L87-L89) or also set it in [here](https://github.com/ankitpokhrel/tus-php/blob/master/src/Tus/Client.php#L112-L113)?

Comment (samundra) [reply src/Tus/Client.php]: I agree with @ankitpokhrel `getMetaData` should only return metadata and should not make any additional changes to internal data.

Like mentioned before, we can add `filename` to metadata if it is needed afterwards in some parts of the application. Perhaps, as mentioned [this](https://github.com/ankitpokhrel/tus-php/blob/master/src/Tus/Client.php#L87-L89) could be suitable place. Or better if we have `setMetadata` we can add `filename` there.

File: tests/Tus/ClientTest.php [hunk 2]
```diff
@@ -117,6 +117,52 @@ public function it_sets_and_gets_filename()
...
+            'filetype' => base64_encode('video/mp4')
+        ], $this->tusClient->getMetadata());
+
+        $this->tusClient->removeMetadata('filetype');
+        $this->assertEquals([
+            'filename' => base64_encode('file.mp4'),
+        ], $this->tusClient->getMetadata());
+
+        // $this->assertNull($this->tusClient->getFileName());
+        // $this->assertInstanceOf(TusClient::class, $this->tusClient->setFileName('file.txt'));
```
Comment (samundra) [inline tests/Tus/ClientTest.php]: Any reason why these are commented?

Comment (SamsamBabadi) [reply tests/Tus/ClientTest.php]: There no reason, it's my mistake.

File: src/Tus/Client.php (hunk 1 above)
Comment (SamsamBabadi) [reply src/Tus/Client.php]: It is better to can override `filename` with `setMetadata` or `addMetadata`, so [this](https://github.com/ankitpokhrel/tus-php/blob/master/src/Tus/Client.php#L87-L89) is better place I think.

**Comment by SamsamBabadi** [comment]
I think everything is OK and you can accept this.

@samundra @ankitpokhrel

**Comment by ankitpokhrel** [comment]
Thanks @SamsamBabadi !

**Pull request merged**